cmake/options.cmake

```diff
@@ -17,6 +17,8 @@ option(BUILD_SHARED_LIBS "Whether to build shared libraries (like *.dll or *.so)
 
 option(ADD_HEADERS_AS_SOURCES "Whether to add headers as sources of a target or not. This is needed for some IDEs which wouldn't detect headers properly otherwise")
 
+option(OSQUERY_NO_DEBUG_SYMBOLS "Whether to build without debug symbols or not, even if a build type that normally have them has been selected")
```

Comment:
In the past we had tons of options with the `OSQUERY_` prefix. I do not think this was good practice. Is there any problem with naming this `NO_DEBUG_SYMBOLS`?

Reply:
Well in the past there were many variables coming from different sources (set() inside CMake, environment variables, options, etc), so there was a mix of them.
Now they are very few and in general they will only be for CMake usage.
Also, the prefix is there because if we have to include other projects as submodules, we have to avoid naming collision.